pkg/apis/topology/v1alpha1/hypernode_types.go

```diff
@@ -0,0 +1,160 @@
+// +kubebuilder:printcolumn:name="NodeCount",type=integer,JSONPath=`.status.nodeCount`
+// +kubebuilder:printcolumn:name="Age",type=date,JSONPath=`.metadata.creationTimestamp`
+
+// HyperNode represents a collection of nodes sharing similar network topology or performance characteristics.
+type HyperNode struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec defines the desired configuration of the HyperNode.
+	// +optional
+	Spec HyperNodeSpec `json:"spec"`
+
+	// Status provides the current state of the HyperNode.
+	// +optional
+	Status HyperNodeStatus `json:"status,omitempty"`
+}
+
+// MemberType represents the member type, valid values are "Node" and "HyperNode".
+// +kubebuilder:validation:Enum=Node;HyperNode
+type MemberType string
+
+const (
+	// MemberTypeNode means the member type is a node.
+	MemberTypeNode MemberType = "Node"
+	// memberTypeHyperNode means the member type is a hyperNode.
+	memberTypeHyperNode MemberType = "HyperNode"
+)
+
+// HyperNodeSpec defines the desired state of a HyperNode.
+type HyperNodeSpec struct {
+	// Tier categorizes the performance level of the HyperNode.
+	// +required
+	Tier string `json:"tier,omitempty"`
+
+	// Members defines a list of node groups or individual nodes included in the HyperNode.
+	// +optional
+	Members []MemberSpec `json:"members,omitempty"`
+}
+
+// MemberSpec represents a specific node or a hyperNodes in the hyperNode.
+type MemberSpec struct {
+	// Type specifies the member type.
+	// +required
+	Type MemberType `json:"type,omitempty"`
+
+	// Selector defines the selection rules for this member.
+	// +optional
+	Selector MemberSelector `json:"selector,omitempty"`
+}
+
+// MemberSelectorType specifies the selection method, valid values are "Exact" and "Regex".
+// +kubebuilder:validation:Enum=Exact;Regex
+type MemberSelectorType string
+
+const (
+	// ExactMatchMemberSelectorType matches nodes exactly by name.
+	ExactMatchMemberSelectorType MemberSelectorType = "Exact"
+
+	// RegexMatchMemberSelectorType matches nodes based on a regular expression.
+	RegexMatchMemberSelectorType MemberSelectorType = "Regex"
+)
+
+// MemberSelector defines the criteria for selecting nodes.
+//
+// Example for Exact match:
+//
+//	members:
+//	- selector:
+//	    type: Exact
+//	    exactMatch:
+//	      name: "node1"
+//
+// Example for Regex match:
+//
+//	members:
+//	- selector:
+//	    type: Regex
+//	    regexMatch:
+//	      pattern: "^node-[0-9]+$"
+//
+// +kubebuilder:validation:XValidation:rule="self.type == 'Exact' ? has(self.exactMatch) : true",message="ExactMatch must be specified when Type is 'Exact'"
+// +kubebuilder:validation:XValidation:rule="self.type == 'Regex' ? has(self.regexMatch) : true",message="RegexMatch must be specified when Type is 'Regex'"
+type MemberSelector struct {
+	// Type specifies the selection method (Exact or Regex).
+	// +required
+	Type MemberSelectorType `json:"type"`
```

Comment:
We don't really need a `Type` for `ExactMatch` vs. `RegexMatch`?

We can simply check which is not nil, and prevent users from indicating both.

Reply:
done